src/toolchain/toolchain.ts

```diff
@@ -116,9 +116,17 @@ export class SwiftToolchain {
                 return path.dirname(path.dirname(path.dirname(swiftc)));
             }
             default: {
-                const { stdout } = await execFile("which", ["swiftc"]);
-                const swiftc = stdout.trimEnd();
-                return path.dirname(path.dirname(path.dirname(swiftc)));
+                // use `type swiftc` to find `swiftc`. Run inside /bin/sh to ensure
+                // we get consistent output as different shells output a different
+                // format. Tried running with `-p` but that is not available in /bin/sh
+                const { stdout } = await execFile("/bin/sh", ["-c", "type swiftc"]);
+                const swiftcMatch = /^swiftc is (.*)$/.exec(stdout);
+                if (swiftcMatch) {
+                    const swiftc = swiftcMatch[1];
+                    vscode.window.showInformationMessage("swiftc: ${swiftc}");
```

Comment:
Why do we need a message here (it's diverged with other platforms)?

Instead, maybe we can add this information to log output?

Reply:
damn that's debug info. I was running inside a remote container to test different variants of linux. I'll remove

Reply:
A few more comments:
1. Why was it `swiftc` instead of `swift` here? We never call `swiftc` directly in the plugin.
	In the rare case where the toolchain is hidden from `PATH` (the support is broken and I've recently submitted some pitches), `swift` and `swiftc` should both be exposed.
2. I guess we should also handle cases when `swift` cannot be found on Darwin and Windows.
3. Why not throw an error when `swift` cannot be found? The plugin will always crash then and the users are not informed of the actual reason.

Reply:
Not sure why we chose swiftc over swift, don't think its a huge issue. If swift doesn't exist the extension will already fail to load, so no need to do another check here. The main issue here is if `type swiftc` fails, because we are running on some version of linux that hasn't been tested which produces different output from `type`.

Reply:
Actually I'll do the throwing of the error, means we don't have an optional toolchain